pkg/ottl/ottlfuncs/func_contains.go

```diff
@@ -0,0 +1,51 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type ContainsArguments[K any] struct {
+	Target ottl.PSliceGetter[K]
+	Item   ottl.Getter[K]
+}
+
+func NewContainsFactory[K any]() ottl.Factory[K] {
+	return ottl.NewFactory("Contains", &ContainsArguments[K]{}, createContainsFunction[K])
+}
+
+func createContainsFunction[K any](_ ottl.FunctionContext, oArgs ottl.Arguments) (ottl.ExprFunc[K], error) {
+	args, ok := oArgs.(*ContainsArguments[K])
+
+	if !ok {
+		return nil, fmt.Errorf("ContainsFactory args must be of type *ContainsArguments[K]")
+	}
+
+	return contains(args.Target, args.Item), nil
+}
+
+func contains[K any](target ottl.PSliceGetter[K], itemGetter ottl.Getter[K]) ottl.ExprFunc[K] {
+	return func(ctx context.Context, tCtx K) (any, error) {
+		slice, sliceErr := target.Get(ctx, tCtx)
+		if sliceErr != nil {
+			return nil, sliceErr
+		}
+		item, itemErr := itemGetter.Get(ctx, tCtx)
+		if itemErr != nil {
+			return nil, itemErr
+		}
+
+		for i := 0; i < slice.Len(); i++ {
+			val := slice.At(i).AsRaw()
+			if val == item {
+				return true, nil
+			}
```

Comment:
Can this return incomparable types? (e.g. slices or maps)

Can we handle the various types explicitly?

Reply:
I think it can return incomparable types, let me think how to do it explicitly. Do you think about such solution?

```
		for i := 0; i < slice.Len(); i++ {
			val := slice.At(i).AsRaw()

			switch val.(type) {
			case string:
				if s, ok := item.(string); ok && s == val {
					return true, nil
				}
			}
		}
```

Reply:
I think we can check if the value is a complex type like a slice or map and perform some kind of deep equality check where necessary. For any primitive type, we can just compare them directly; I don't think we have to go as far as casting like this since two different primitive types will trivially fail the equality check.